### ISSUE TITLE
Fix ref resolution during response validation

### ISSUE THREAD

**Issue by analogue** [Yelp/bravado-core#82]
Ref resolution during response validation sometimes fails because of where the validation starts. For refs to resolve 100% of the time, the scope maintained by jsonschema's RefResolver has to be rooted at `/` in swagger.json, not at the response_spec fragment that is currently passed to the `validate_response(..)` function.  

Luckily, the full scope is available as `x-scope` annotations in the spec so the `ref` validator just needs to be customized to use the `x-scope` annotation if it is available.

This is quite difficult to explain, so an example is probably useful to compare working and broken scopes for the RefResolver for an internal use case:

Broken RefResolver scope:
```
['file:///nail/home/spatel/pg/services/biz_app/api_docs/swagger.json']
```

Working RefResolver scope:
```
[
'file:///nail/home/spatel/pg/services/biz_app/api_docs/swagger.json',
'file:///nail/home/spatel/pg/services/biz_app/api_docs/tags/account.json#/paths/account.photo.v1',
'file:///nail/home/spatel/pg/services/biz_app/api_docs/tags/account.json#/definitions/AccountPhotoResponse'
]
```

**Comment by analogue** [comment]
fyi, @sjaensch

**Comment by analogue** [comment]
Fixed in 4.2.1
